lib/rewrite/rewrite-expr-grammar.ym

```diff
@@ -158,9 +159,14 @@ rewrite_subst_opt
 
 rewrite_expr_opts
         : rewrite_expr_opt rewrite_expr_opts
+        | rewrite_set_opt rewrite_expr_opts
         |
         ;
 
+rewrite_set_opt
+        : template_option
+        ;
+
 rewrite_expr_opt
```

Comment:
Usually these kind of grammars, we only use a single recursion and the _opt rule includes all potential options, set and generic expr alike.